tests/unittests/core/test_experiment.py

```diff
@@ -448,7 +448,6 @@ def test_algorithm_config_with_just_a_string(self, exp_config):
         new_config['algorithms'] = 'dumbalgo'
         exp = Experiment('supernaedo3')
         exp.configure(new_config)
-        new_config['status'] = 'pending'
```

Comment:
When should the status change effectively to 'pending'?

Reply:
When the maximum number of trials is higher than current number of trials and not algo.is_done. One or the other condition is enough to force 'completed'.